src/natcap/invest/urban_nature_access.py

```diff
@@ -547,6 +623,148 @@ def execute(args):
+    """Reclassify the admin units and multiply by greenspace supply.
+
+    Args:
+        admin_units_raster_path (string): The path to a raster of integers
+            identifying which admin unit a pixel belongs to.
+        reclassification_map (dict): A dict mapping integer admin unit IDs to
+            float population proportions (values 0-1) for a given population
+            group.
+        supply_raster_path (string): A string path to a raster of greenspace
+            supply values for the total population.
+        target_raster_path (string): The string path to where the resulting
+            supply-to-group raster should be written.
+
+    Returns:
+        ``None``
+    """
+    pygeoprocessing.reclassify_raster(
+        (admin_units_raster_path, 1), reclassification_map, target_raster_path,
+        gdal.GDT_Float32, FLOAT32_NODATA)
+
+    pop_group_raster = gdal.OpenEx(target_raster_path,
+                                   gdal.GA_Update | gdal.OF_RASTER)
+    pop_group_band = pop_group_raster.GetRasterBand(1)
+    pop_group_nodata = pop_group_band.GetNoDataValue()
+    supply_raster = gdal.OpenEx(supply_raster_path,
+                                gdal.GA_ReadOnly | gdal.OF_RASTER)
+    supply_band = supply_raster.GetRasterBand(1)
+    supply_nodata = supply_band.GetNoDataValue()
+    for block_info in pygeoprocessing.iterblocks((target_raster_path, 1),
+                                                 offset_only=True):
+        pop_group_proportion_block = pop_group_band.ReadAsArray(**block_info)
+        supply_block = supply_band.ReadAsArray(**block_info)
+
+        valid_mask = (
+            ~utils.array_equals_nodata(
+                pop_group_proportion_block, pop_group_nodata) &
+            ~utils.array_equals_nodata(supply_block, supply_nodata))
+        target_block = numpy.full(supply_block.shape, FLOAT32_NODATA,
+                                  dtype=numpy.float32)
+        target_block[valid_mask] = (
+            pop_group_proportion_block[valid_mask] * supply_block[valid_mask])
+        pop_group_band.WriteArray(
+            target_block, xoff=block_info['xoff'], yoff=block_info['yoff'])
+
+    pop_group_band = None
+    pop_group_raster = None
+    supply_band = None
+    supply_raster = None
+
+
+def _read_field_from_vector(vector_path, key_field, value_field):
+    """Read a field from a vector's first layer.
+
+    Args:
+        vector_path (string): The string path to a vector.
+        key_field (string): The string key field within the vector.
+            ``key_field`` must exist within the vector at ``vector_path``.
+            ``key_field`` is case-sensitive.
+        value_field (string): The string value field within the vector.
+            ``value_field`` must exist within the vector at ``vector_path``.
+            ``value_field`` is case-sensitive.
+
+    Returns:
+        attribute_map (dict): A dict mapping ``key_field`` to
+            ``value_field`` value.
```

Comment:
```suggestion
        attribute_map (dict): A dict mapping each ``key_field`` value to
            the corresponding ``value_field`` value.
```

Reply:
This one is a little tricky because of the duplication of `'value'`, since `key` and `value` are the conventional words for the dictionary accessor and accessed data, respectively.  How about this instead?

```
attribute_map (dict): A dict mapping each ``key_field`` key to 
    the corresponding ``value_field`` value.
```

In case that works, I've added it in 02a52e981

Reply:
that does read better to me, thanks!